templates/bbb.nomad

```diff
@@ -0,0 +1,217 @@
+        cpu = 100
+        memory = 300
+        network {
+          mbits = 1
+          port "bbb_pg" {}
+        }
+      }
+
+      service {
+        name = "bbb-pg"
+        port = "bbb_pg"
+        tags = ["fabio-:${config.port_bbb_pg} proto=tcp"]
+        check {
+          name = "pg_isready"
+          type = "script"
+          command = "/bin/sh"
+          args = ["-c", "pg_isready"]
+          interval = "${check_interval}"
+          timeout = "${check_timeout}"
+        }
+      }
+    }
+
+    task "bbb-redis" {
+      ${ task_logs() }
+      constraint {
+        attribute = "{% raw %}${meta.liquid_volumes}{% endraw %}"
+        operator = "is_set"
+      }
+      ${ shutdown_delay() }
+
+      driver = "docker"
+      config {
+        image = "${config.image('bbb-redis')}"
+        volumes = [
+          "{% raw %}${meta.liquid_volumes}{% endraw %}/bbb/redis/data:/data",
+        ]
+        port_map {
+          redis = 6379
+        }
+        labels {
+          liquid_task = "bbb-redis"
+        }
+      }
+
+      resources {
+        cpu = 500
+        memory = 512
+        network {
+          port "redis" {}
+          mbits = 1
+        }
+      }
+
+      service {
+        name = "bbb-redis"
+        port = "redis"
+        tags = ["fabio-:${config.port_bbb_redis} proto=tcp"]
+
+        check {
+          name = "bbb-redis-alive"
+          initial_status = "critical"
+          type = "tcp"
+          interval = "${check_interval}"
+          timeout = "${check_timeout}"
+        }
+        check_restart {
+          limit = 5
+          grace = "980s"
+        }
+      }
+    }
+
+    task "greenlight" {
+      ${ task_logs() }
+      leader = false
+
+      constraint {
+        attribute = "{% raw %}${meta.liquid_volumes}{% endraw %}"
+        operator = "is_set"
+      }
+
+      affinity {
+        attribute = "{% raw %}${meta.liquid_large_databases}{% endraw %}"
+        value     = "true"
+        weight    = 100
+      }
+
+      driver = "docker"
+      ${ shutdown_delay() }
+
+      template {
+        data = <<-EOF
+        {{- with secret "liquid/bbb/bbb.postgres" }}
+        DATABASE_URL = "postgres://greenlight:{{.Data.secret_key | toJSON }}@{{env "attr.unique.network.ip-address"}}:${config.port_bbb_pg}/greenlight_production"
+        {{- end }}
+        REDIS_URL = "redis://{{ env "attr.unique.network.ip-address"}}:${config.port_bbb_redis}"
+        BIGBLUEBUTTON_ENDPOINT = "${config.bbb_endpoint}"
+        BIGBLUEBUTTON_SECRET = "${config.bbb_secret}"
+        {{- with secret "liquid/bbb/bbb.secret_key_base" }}
+        SECRET_KEY_BASE = {{.Data.secret_key | toJSON }}
+        {{- end }}
+        RELATIVE_URL_ROOT = "/"
+        {{- with secret "liquid/bbb/auth.oauth2" }}
+        OPENID_CONNECT_CLIENT_ID = {{.Data.client_id | toJSON }}
+        OPENID_CONNECT_CLIENT_SECRET = {{.Data.client_secret | toJSON }}
+        {{- end }}
+        OPENID_CONNECT_ISSUER="https://${liquid_domain}"
```

Comment:
what if this is hosted on HTTP (for example tor)?

IIRC webrtc doesn't even start connections without HTTPS, so it might not make sense to put the protocol vars here instead of the hard-coded `https`

Reply:
nice catch, replaced with config.liquid_http_protocol, re-deployed, tested, ok now